### PR TITLE
Bump proxy-init:v2.2.3 and cni-plugin:v1.2.2

### PR DESCRIPTION
**Pull request by alpeb**
https://github.com/linkerd/linkerd2-proxy-init/releases/tag/proxy-init%2Fv2.2.3
https://github.com/linkerd/linkerd2-proxy-init/releases/tag/cni-plugin%2Fv1.2.2

Dependency bumps and updated go to v1.21

### CODE REVIEW

**Review by mateiidavid — APPROVED** [review]

File: charts/linkerd-control-plane/README.md [hunk 1]
```diff
@@ -224,6 +224,7 @@ Kubernetes: `>=1.21.0-0`
 | profileValidator.injectCaFromSecret | string | `""` | Inject the CA bundle from a Secret. If set, the `cert-manager.io/inject-ca-from-secret` annotation will be added to the webhook. The Secret must have the CA Bundle stored in the `ca.crt` key and have the `cert-manager.io/allow-direct-injection` annotation set to `true`. See the cert-manager [CA Injector Docs](https://cert-manager.io/docs/concepts/ca-injector/#injecting-ca-data-from-a-secret-resource) for more information. |
 | profileValidator.keyPEM | string | `""` | Certificate key for the service profile validator. If not provided and not using an external secret then Helm will generate one. |
 | profileValidator.namespaceSelector | object | `{"matchExpressions":[{"key":"config.linkerd.io/admission-webhooks","operator":"NotIn","values":["disabled"]}]}` | Namespace selector used by admission webhook |
+| prometheusUrl | string | `""` | url of external prometheus instance (used for the heartbeat) |
```
Comment (olix0r) [inline charts/linkerd-control-plane/README.md:227]: This seems unrelated to proxyInit?

Comment (mateiidavid) [reply charts/linkerd-control-plane/README.md:227]: It's from a previous change where `bin/helm-docs` wasn't run. Seems to be unavoidable when generating the docs. For the backport _it might be okay_, if the field isn't present in `values.yaml` it will get removed once we bump the charts and re-generate the configs.

@alpeb does that sound right to you?

Comment (alpeb) [reply charts/linkerd-control-plane/README.md:227]: I don't understand why the doc diff wasn't caught by CI in earlier runs, but it looks like this PR puts it back in track.

**Review by olix0r — APPROVED** [review]
